# Outbreak Readiness System

In [20]:
import numpy as np
import pandas as pd
import json
#from dbfread import DBF
import geopandas as gpd
from explore import explore

## 2. Data Understanding

### 2.1. Kenya_hxl.csv dataset

#### 2.1.1. Loading the file

In [17]:
hxl_df = pd.read_csv("./Datasets/kenya_hxl.csv")
hxl_df.head()

,X,Y,osm_id,osm_type,completeness,#loc+amenity,#meta+healthcare,#loc +name,#meta +operator,#geo+bounds+url,...,#meta+is_in_health_zone,#contact +url,addr_housenumber,addr_street,addr_postcode,addr_city,changeset_id,changeset_version,changeset_timestamp,#meta +id
0,NaN,NaN,474439689,way,21.875,hospital,counselling,Community Health Center,MOH,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170715105,5,2025/08/20 09:49:32+00,1e84916be0a9432a8b98d539bc014957
1,NaN,NaN,1224725709,way,15.625,hospital,hospital,JOHATTLAS MEDICAL AND DENTAL CENTER,NaN,microsoft/BuildingFootprints,...,NaN,NaN,NaN,NaN,NaN,NaN,163982188,2,2025/03/23 11:25:46+00,020c4dec61384d65853adb9ee4680784
2,36.887300,-1.254270,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,107380437,1,2021/07/04 08:44:18+00,4ac9577b8e094463b9436f95117a07c1
3,36.887851,-1.246880,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,17201651,1,2013/08/03 09:20:18+00,71c100d49785492283c8c7475c7d172d
4,36.886524,-1.244329,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,Nairobi,99124843,1,2021/02/11 17:31:32+00,3ad1a6e7f46649419f72c447d57ad7c7


#### 2.1.2. Checking the shape of the dataset

In [31]:
hxl_df.shape

(2640, 35)

**kenya_hxl** has **2,640** rows and **35** columns

#### 2.1.3. Checking for missing data

In [39]:
hxl_df.isnull().sum()

X                              721
Y                              721
osm_id                           0
osm_type                         0
completeness                     0
#loc+amenity                    32
#meta+healthcare              1231
#loc +name                     118
#meta +operator               2051
#geo+bounds+url               2464
#meta +speciality             2461
#meta +operator_type          1430
#contact +phone               2543
#status+operational_status    1922
#access +hours                2045
#capacity +beds               2510
#capacity +staff              2640
#meta +health_amenity_type    2622
#meta+dispensing              2256
#meta+wheelchair              2583
#meta+emergency               2506
#meta+insurance               2632
#meta+water_source            2623
#meta+electricity             2549
#meta+is_in_health_area       2640
#meta+is_in_health_zone       2640
#contact +url                 2637
addr_housenumber              2583
addr_street         

#### 2.1.4. Checking for datatypes

### 2.2. Kenya.csv dataset

#### 2.2.1. Loading the file

In [18]:
kenya_df = pd.read_csv("./Datasets/kenya.csv")
kenya_df.head()

,X,Y,osm_id,osm_type,completeness,amenity,healthcare,name,operator,source,...,is_in_health_zone,url,addr_housenumber,addr_street,addr_postcode,addr_city,changeset_id,changeset_version,changeset_timestamp,uuid
0,NaN,NaN,474439689,way,21.875,hospital,counselling,Community Health Center,MOH,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170715105,5,2025/08/20 09:49:32+00,1e84916be0a9432a8b98d539bc014957
1,NaN,NaN,1224725709,way,15.625,hospital,hospital,JOHATTLAS MEDICAL AND DENTAL CENTER,NaN,microsoft/BuildingFootprints,...,NaN,NaN,NaN,NaN,NaN,NaN,163982188,2,2025/03/23 11:25:46+00,020c4dec61384d65853adb9ee4680784
2,36.887300,-1.254270,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,107380437,1,2021/07/04 08:44:18+00,4ac9577b8e094463b9436f95117a07c1
3,36.887851,-1.246880,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,17201651,1,2013/08/03 09:20:18+00,71c100d49785492283c8c7475c7d172d
4,36.886524,-1.244329,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,Nairobi,99124843,1,2021/02/11 17:31:32+00,3ad1a6e7f46649419f72c447d57ad7c7


### 2.3. Kenya_hxl.geojson dataset

#### 2.3.1. Loading the file

In [30]:
with open("./Datasets/kenya_hxl.geojson", "r") as file:
    hxl_json = json.load(file)

### 2.4. Kenya.geojson

#### 2.4.1. Loading the file

In [27]:
with open("./Datasets/kenya.geojson", "r") as file:
    kenya_json = json.load(file)

### 2.5. Kenya.dbf

#### 2.5.1. Loading the file

In [ ]:
dbf_df = gpd.read_file("/workspaces/Outbreak-Readiness-System/Datasets/kenya-shapefiles/shapefiles/kenya.dbf")
dbf_df.head()

,osm_id,osm_type,completene,amenity,healthcare,name,operator,source,speciality,operator_t,...,url,addr_house,addr_stree,addr_postc,addr_city,changeset_,changese_1,changese_2,uuid,geometry
0,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,107380437.0,1.0,2021-07-04,4ac9577b8e094463b9436f95117a07c1,POINT (36.8873 -1.25427)
1,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,17201651.0,1.0,2013-08-03,71c100d49785492283c8c7475c7d172d,POINT (36.88785 -1.24688)
2,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,NaN,private,...,None,NaN,NaN,NaN,Nairobi,99124843.0,1.0,2021-02-11,3ad1a6e7f46649419f72c447d57ad7c7,POINT (36.88652 -1.24433)
3,6244568075,node,9.375,hospital,NaN,Ngelani Dispensary,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,66740351.0,1.0,2019-01-29,a5fc107df3564044bcd9e39ca4180181,POINT (38.27305 -1.38409)
4,12238169033,node,21.875,pharmacy,NaN,kwa grace chemist,NaN,survey,NaN,private,...,None,NaN,NaN,NaN,NaN,159324175.0,2.0,2024-11-19,293d5095a6db4ce9bac452eecdc188f5,POINT (36.86421 -1.31251)
